Implemented the products API.

TomixJasina opened the pull request:
Added maximum products per page to the service config.

Corrected product information in the output schema in the products API.

Implemented the products API.

TomixJasina committed:
Added maximum products per page to the service config.

Corrected product information in the output schema in the products API.

Implemented the products API.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/BetterCorp/WISPStock?pullRequest=1) <br/>All committers have signed the CLA.

mrinc commented on src/plugins/service-wispstock/views/betterportal/products/index.ts:
```diff
@@ -49,10 +53,49 @@ export const ViewHandlers: Record<T_SUPPORTED_THEMES, BetterPortalUIView<ViewDef
+    .whereNotEquals("enabled", true);
+
+  if (search) {
+    query1 = query1.whereEquals("title", search);
```
We need a more fuzzy search here

mrinc replied:
Also should match on SKU as well

mrinc commented on src/plugins/service-wispstock/views/betterportal/products/index.ts:
```diff
@@ -49,10 +53,49 @@ export const ViewHandlers: Record<T_SUPPORTED_THEMES, BetterPortalUIView<ViewDef
+  const search = props.query.search;
+
+  let query1 = session
+    .query<ProductListItem>({ collection: "products" })
```
We should probs add the collection name as a config variable (secConfig)?

mrinc commented on src/plugins/service-wispstock/views/betterportal/products/index.ts:
```diff
@@ -49,10 +53,49 @@ export const ViewHandlers: Record<T_SUPPORTED_THEMES, BetterPortalUIView<ViewDef
+  ViewDefinition
+> = async (props) => {
+  const session = props.context.RavenDB.openSession();
+
```
Please group these variables (general cleanup)

TomixJasina committed:
Added global collections names constant

Added global RavenDB query quick reference for collections

Fixed wildcard query search

Sorted variables in DataHandler

General fixes

TomixJasina committed:
Merge branch 'master' of https://github.com/BetterCorp/WISPStock

TomixJasina committed:
- Updated product information in the products API output schema.

- Updated styling for the product card.

- Added a new CSS file for product card styling.

- Included product images and dummy images.

- Added a link to the product details page.

mrinc commented on src/plugins/service-wispstock/plugin.ts:
```diff
@@ -64,6 +65,8 @@ export class Plugin
   runBeforePlugins?: string[] | undefined;
   runAfterPlugins?: string[] | undefined;
+  public get Config() { return this.config; }
+  public get pcwd(): string { return this.pluginCwd; }
```
Shouldn't we change this property name to match the get above for Config? `PluginCWD`?

mrinc commented on src/plugins/service-wispstock/views/betterportal/products/materio1.tsx:
```diff
@@ -1,17 +1,113 @@
+import path from "path";
 
 export const Component: BetterPortalUIView<ViewDefinition> = (props) => {
+  const page = props.outboundData.page,
```
Don't create extra variables unless you need too - since you can ref props directly.

TomixJasina committed:
Deleted unnecessary variable
Rename property name
Added styling of product card (WIP)

TomixJasina committed:
Deleted img of product
Added styling
